src/libraries/Common/tests/TestUtilities/System/PlatformDetection.cs

```diff
@@ -143,7 +143,12 @@ public static int SlowRuntimeTimeoutModifier
         public static bool IsStartingProcessesSupported => !IsiOS && !IstvOS;
 
         public static bool IsSpeedOptimized => !IsSizeOptimized;
-        public static bool IsSizeOptimized => IsBrowser || IsWasi || IsAndroid || IsAppleMobile;
+        public static bool IsSizeOptimized => s_linqIsSizeOptimized.Value;
```

Comment:
Do we want to rename this property to `IsLinqSizeOptimized ` or `IsSizeOptimizedLinq` since it only applies to Linq now?

Reply:
It looks like there are some places in the codebase that use IsSpeedOptimized and that are not related to LINQ. We need to re-evaluate those as part of this. e.g.
https://github.com/dotnet/runtime/blob/d20c35e05c16dd2ec9cddd4259738cc02228d425/src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs#L12-L14

Reply:
Good catch. Renamed to IsLinqXXXOptimized and replaced the non-linq one with IsNotMobile since it's equivalent and we use Mobile as "slow".